yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -324,7 +324,7 @@ export class AztecNodeService implements AztecNode {
    */
   public async getL1ToL2MessageAndIndex(messageKey: Fr): Promise<L1ToL2MessageAndIndex> {
     // todo: #697 - make this one lookup.
-    const index = (await this.findLeafIndex(MerkleTreeId.L1_TO_L2_MESSAGES_TREE, messageKey.toBuffer()))!;
+    const index = (await this.findLeafIndex(MerkleTreeId.L1_TO_L2_MESSAGES_TREE, messageKey))!;
```

Comment:
wait hold up above uses buffer and this uses FR?

Reply:
I am AFK already so I merged it and I will consider addressing this tomorrow.